fxa-auth-db-server/test/backend/db_tests.js

```diff
@@ -1635,38 +1636,49 @@ module.exports = function(config, DB) {
                   createSession(sessionId2, session2),
                   createSession(sessionId3, session3)
                 ])
+                // Don't paralleize these, the order of them matters
+                // because they record timestamps in the db.
                 .then(function () {
-                  return P.all([
-                    insert(uid1, addr1, evB, sessionId1),
-                    insert(uid1, addr1, evA, sessionId2),
-                    insert(uid1, addr2, evA, sessionId3),
-                    insert(uid2, addr1, evA, hex32())
-                  ])
+                  return insert(uid1, addr1, evA, sessionId2).then(P.delay.bind(1))
+                })
+                .then(function () {
+                  return insert(uid1, addr1, evB, sessionId1).then(P.delay.bind(1))
+                })
+                .then(function () {
+                  return insert(uid1, addr1, evC).then(P.delay.bind(1))
+                })
+                .then(function () {
+                  return insert(uid1, addr2, evA, sessionId3).then(P.delay.bind(1))
+                })
+                .then(function () {
+                  return insert(uid2, addr1, evA, hex32())
```

Comment:
Not particularly important, and maybe it's not as clean as I think it is, but I quite like `Array.reduce` for serialising repetitive promise chains like this. In this case, perhaps something like:

```js
[
  { uid: uid1, address: addr1, event: evA, session: sessionId2 },
  { uid: uid1, address: addr1, event: evB, session: sessionId1 },
  { uid: uid1, address: addr1, event: evC },
  { uid: uid1, address: addr2, event: evA, session: sessionId3 },
  { uid: uid2, address: addr1, event: evA, session: hex32() }
].reduce((promise, record) => {
  return promise
    .then(() => insert(record.uid, record.address, record.event, record.session))
    .then(() => P.delay(1))
}, P.resolve())
```